proto-public/pbauth/v1alpha1/traffic_permissions.proto

```diff
@@ -0,0 +1,126 @@
+syntax = "proto3";
+
+package hashicorp.consul.auth.v1alpha1;
+
+message TrafficPermissions {
+  // destination is a configuration of the destination proxies
+  // where these traffic permissions should apply.
+  Destination destination = 1;
+
+  // Action can be either allow or deny for the entire object. It will default to allow.
+  //
+  // If action is allow,
+  // we will allow the connection if one of the rules in Rules matches, in other words, we will deny
+  // all requests except for the ones that match Rules. If Consul is in default allow mode, then allow
+  // actions have no effect without a deny permission as everything is allowed by default.
+  //
+  // If action is deny,
+  // we will deny the connection if one of the rules in Rules match, in other words,
+  // we will allow all requests except for the ones that match Rules. If Consul is default deny mode,
+  // then deny permissions have no effect without an allow permission as everything is denied by default.
+  //
+  // Action unspecified is reserved for compatibility with the addition of future actions.
+  Action action = 2;
+
+  // permissions is a list of permissions to match on.
+  // They are applied using OR semantics.
+  repeated Permission permissions = 3;
+}
+
+message NamespaceTrafficPermissions {
+  Action action = 1;
+  repeated Permission permissions = 2;
+}
+
+message PartitionTrafficPermissions {
+  Action action = 1;
+  repeated Permission permissions = 2;
+}
+
+// Destination contains the name or name-prefix of the WorkloadIdentity.
+// The WorkloadIdentity resource must
+// be in the same tenancy as the TrafficPermissions resource.
+message Destination {
+  string identity_name = 1;
+  string identity_prefix = 2;
+}
+
+enum Action {
+  ACTION_UNSPECIFIED = 0;
+  ACTION_DENY = 1;
+  ACTION_ALLOW = 2;
```

Comment:
As I commented above, this should just have two actions allow and deny with default being "allow". This defaulting is intentional in this case because we want to have a more secure option be the default (same as envoy).